### PR TITLE
[TF:MLIR] Convert FusedBatchNorm to FusedBatchNormV3 in prepare TFLite pass

### PR DESCRIPTION
**Pull request by ahmedsabie**

### CODE REVIEW

**Review by smit-hinsu — COMMENTED** [review]
Have a few comments but otherwise looks correct.

**Review by smit-hinsu — DISMISSED** [review]

**Comment by gbaned** [comment]
@ahmedsabie  Can you please check @smit-hinsu's comments and keep us posted ? Thanks!

**Review by smit-hinsu — DISMISSED** [review]

File: tensorflow/compiler/mlir/lite/transforms/prepare_tf.cc [hunk 1]
```diff
@@ -727,6 +727,33 @@ struct ConvertTFBroadcastTo : public RewritePattern {
...
+  explicit ConvertFusedBatchNorm(MLIRContext *context)
+      : RewritePattern(TF::FusedBatchNormOp::getOperationName(), 1, context) {}
+
+  LogicalResult matchAndRewrite(Operation *op,
+                                PatternRewriter &rewriter) const override {
+    auto tf_fused_batch_norm_op = cast<TF::FusedBatchNormOp>(op);
+
+    // FusedBatchNormV3 expects a 5th output reserve_space_3,
+    // but the output is unused; it doesn't matter what we pass there.
+    rewriter.replaceOpWithNewOp<TF::FusedBatchNormV3Op>(
```
[the remark this answers is not in the document]
Comment (smit-hinsu) [reply tensorflow/compiler/mlir/lite/transforms/prepare_tf.cc]: This is still an issue.

**Review by smit-hinsu — COMMENTED** [review]
Please take a look at this.

File: tensorflow/compiler/mlir/lite/transforms/prepare_tf.cc [hunk 2]
```diff
@@ -746,8 +746,13 @@ struct ConvertFusedBatchNorm : public OpRewritePattern<TF::FusedBatchNormOp> {
                              tf_fused_batch_norm_op.getAttrs());
     Operation *tf_fused_batch_norm_op_v3 = rewriter.createOperation(new_state);
 
-    rewriter.replaceOp(tf_fused_batch_norm_op,
-                       tf_fused_batch_norm_op_v3->getResults());
```
Comment (smit-hinsu) [inline tensorflow/compiler/mlir/lite/transforms/prepare_tf.cc]: We can still use replaceOp by using take_front or drop_back method on getResults().

**Review by smit-hinsu — APPROVED** [review]

**Pull request merged**